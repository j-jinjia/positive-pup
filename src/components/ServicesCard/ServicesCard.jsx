import "./ServicesCard.scss";
import vectorBlack from "../../assets/svgs/Vector-black.svg";
import vectorWhite from "../../assets/svgs/Vector-white.svg";
import { Link } from "react-router-dom";

const ServicesCard = ({ title, description }) => {
  return (
    <div className="services-card">
      <h2 className="services-card__title">{title}</h2>
      <h3>{description}</h3>
      <Link to="/services">
        <div className="services-card__button">
          <img
            className="services-card__vector services-card__vector--black"
            src={vectorBlack}
            alt=""
          />
          {/* Put a second chevron here of a different colour with display=none. We can use this to change the colour when we add hover later */}
          <img
            className="services-card__vector services-card__vector--white"
            src={vectorWhite}
            alt=""
          />
        </div>
      </Link>
    </div>
  );
};

export default ServicesCard;
