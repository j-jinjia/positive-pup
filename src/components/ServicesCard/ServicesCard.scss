@use "../../styles/variables" as *;

.services-card {
  background-color: #e4f5f3;
  max-width: 23.125rem;
  max-height: 24.125rem;
  filter: drop-shadow(0px 20px 40px rgba(0, 0, 0, 0.04));
  border-radius: 0.4375rem;
  padding: 2.5rem;
  margin-bottom: 2rem;

  &__title {
    margin-bottom: 1rem;
  }
  &__button {
    border: 1px solid black;
    border-radius: 100%;
    height: 2.8rem;
    width: 2.8rem;
    display: flex;
    align-items: center;
    justify-content: center;
    margin-top: 1rem;
  }
}
