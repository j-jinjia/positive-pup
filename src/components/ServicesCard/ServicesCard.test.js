import { customRender } from "../../utils/testUtils";
import ServicesCard from "./ServicesCard";

it("should match ServicesCard component snapshot", () => {
  const { container } = customRender(<ServicesCard />);

  expect(container).toMatchSnapshot();
});