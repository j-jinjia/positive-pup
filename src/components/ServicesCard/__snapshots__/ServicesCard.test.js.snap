// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match ServicesCard component snapshot 1`] = `
<div>
  <div
    class="services-card"
  >
    <h3
      class="services-card__title"
    >
      Online Courses
    </h3>
    <p
      class="services-card__text"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit.
    </p>
    <a
      aria-label="link to services filtered for Online Courses"
      class="services-card__button"
      href="/services/Online Courses"
    >
      <img
        alt=""
        class="services-card__vector services-card__vector--black"
        src="Vector-black.svg"
      />
    </a>
    <img
      alt="dog"
      class="services-card__img"
      src="image.png"
    />
  </div>
</div>
`;
